Aplib.Core/Tactics/AnyOfTactic.cs

```diff
@@ -32,19 +32,22 @@ public AnyOfTactic(List<Tactic> subTactics)
         /// </summary>
         /// <param name="subTactics">The list of sub-tactics.</param>
         /// <param name="guard">The guard condition.</param>
-        public AnyOfTactic(List<Tactic> subTactics, Func<bool> guard) : this(subTactics) => Guard = guard;
+        public AnyOfTactic(Func<bool> guard, params Tactic[] subTactics) : this(subTactics) => Guard = guard;
 
         /// <inheritdoc/>
-        public override List<PrimitiveTactic> GetFirstEnabledActions()
+        public override Action? GetAction()
         {
-            List<PrimitiveTactic> primitiveTactics = new();
+            List<Action> actions = new();
 
             foreach (Tactic subTactic in SubTactics)
             {
-                primitiveTactics.AddRange(subTactic.GetFirstEnabledActions());
+                Action? action = subTactic.GetAction();
+
+                if (action is not null)
+                    actions.Add(action);
             }
 
-            return primitiveTactics;
+            return actions[ThreadSafeRandom.Next(actions.Count)];
```

Comment:
The System namespace already has a thread-safe static Random instance:
```suggestion
            return actions[Random.Shared.Next(actions.Count)];
```

Reply:
I thought Random.Shared comes in .NET 6 but I can check tomorrow.

Reply:
Indeed, `Random.Shared` is available from .NET 6. https://learn.microsoft.com/en-us/dotnet/api/system.random.shared?view=net-8.0

Reply:
You are right